File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque
input = sys.stdin.readline

def act(q:deque, com, num):
    if com == 'push':
        q.append(num)
    if com == 'pop':
        if q:
            print(q.popleft())
        else:
            print(-1)
    if com == 'size':
        print(len(q))
    if com == 'empty':
        if not q:
            print(1)
        else:
            print(0)
    if com == 'front':
        if q:
            print(q[0])
        else:
            print(-1)
    if com == 'back':
        if q:
            print(q[-1])
        else:
            print(-1)

q = deque()
for _ in range(int(input())):
    com, *num = input().rstrip().split()
    if num:
        num = int(num[0])
    act(q, com, num)

File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static void main(String[] args) throws IOException {
        Queue<String> q = new LinkedList<>();
        int n = Integer.parseInt(br.readLine());
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        while (n-- > 0) {
            st = new StringTokenizer(br.readLine());
            switch (st.nextToken()) {
                case "push":
                    q.offer(st.nextToken());
                    break;
                case "pop":
                    sb.append(q.isEmpty() ? "-1" : q.poll()).append("\n");
                    break;
                case "size":
                    sb.append(q.size()).append("\n");
                    break;
                case "empty":
                    sb.append(q.isEmpty() ? "1\n" : "0\n");
                    break;
                case "front":
                    sb.append(q.isEmpty() ? "-1" : q.peek()).append("\n");
                    break;
                case "back":
                    sb.append(q.isEmpty() ? "-1" : ((LinkedList<String>) q).peekLast()).append("\n");
                    break;
            }
        }
        System.out.println(sb);
    }
}
